src/cascadia/TerminalSettingsEditor/PreviewConnection.cpp

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "PreviewConnection.h"
+
+using namespace ::winrt::Microsoft::Terminal::TerminalConnection;
+using namespace ::winrt::Windows::Foundation;
+namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
+{
+    PreviewConnection::PreviewConnection() noexcept
+    {
+    }
+
+    void PreviewConnection::Start() noexcept
+    {
+    }
+
+    void PreviewConnection::WriteInput(hstring const& data)
+    {
+        std::wstringstream prettyPrint;
+        for (const auto& wch : data)
+        {
+            if (wch < 0x20)
+            {
+                prettyPrint << L"^" << gsl::narrow_cast<wchar_t>(wch + 0x40);
+            }
+            else if (wch == 0x7f)
+            {
+                prettyPrint << L"0x7f";
+            }
+            else
+            {
+                prettyPrint << wch;
+            }
+        }
+        _TerminalOutputHandlers(prettyPrint.str());
+    }
+
+    void PreviewConnection::Resize(uint32_t /*rows*/, uint32_t /*columns*/) noexcept
+    {
+    }

```

Comment:
Reflow will be impossible. On resize, the connection should simply clear the terminal and _re-print the entire preview text_.

Reply:
The reason for this is that resize is a destructive operation when content is pushed off the top of the screen.

Reply:
> This method still needs to be implemented for that right?

Ah it actually doesn't! The connection doesn't need to do anything

Reply:
Wait, no.. on resize, the connection must re-print the entire preview text. Resize with reflow *destroys* the content that goes off the top of the screen, and the connection needs to clear the screen and put it back. Now, since we never resize the control this isn't going to be a problem... but do keep it in mind.